lib/features/surveydetails/ui/survey_details_screen.dart

```diff
@@ -78,33 +79,31 @@ class SurveyDetailsScreenState extends ConsumerState<SurveyDetailsScreen> {
     final SurveyDetailsUiModel uiModel = ref.watch(
       surveyDetailsViewModelProvider,
     );
+    final isLoading = uiModel.isLoading;
     final SurveyDetailsModel? survey = uiModel.surveyDetails;
 
+    if (isLoading) {
+      return Container(
+        color: Colors.black,
+        child: Center(child: CircularProgressIndicator()),
+      );
+    }
```

Comment:
@eeeeaa we can use the extension method `uiModel.maybeWhen` to return the widget for the corresponding state here. Ex:
```dart
return uiModel.maybeWhen((surveyDetails, surveyQuestions, isLoading) {
        if (isLoading) {
             return ....
        }
    }
```